bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/handler/HomematicThingHandler.java

```diff
@@ -140,8 +140,7 @@ private void doInitializeInBackground() throws GatewayNotAvailableException, Hom
             loadHomematicChannelValues(channel);
             for (HmDatapoint dp : channel.getDatapoints()) {
                 if (dp.getParamsetType() == HmParamsetType.MASTER) {
-                    config.put(MetadataUtils.getParameterName(dp),
-                            dp.isEnumType() ? dp.getOptionValue() : dp.getValue());
+                    config.put(MetadataUtils.getParameterName(dp), getValueForConfiguration(dp));
```

Comment:
As getValueForConfiguration can retrun null, I am not sure whether this config entry should be set to null, or not set at all ?
I am even not sure if this is valid to set a null value for a config parameter.

Reply:
The only case under which `getValueForConfiguration()` returns null is if the datapoint holds a `null` value. In that case I'd say that writing `null` into the config is correct.
The places I checked that care about the keys present in the map (those that call `containsKey()` and `keySet()` seem to be able to deal with null values being present. Not sure about e.g. the UI, but given the previous code put `null` values in this case (value being `null`) as well, I guess it should be fine.